components/brave_rewards/content/BUILD.gn

```diff
@@ -65,7 +65,7 @@ static_library("browser") {
     "//url",
   ]
 
-  public_deps = [ "//brave/components/brave_rewards/core:headers" ]
+  public_deps = [ "//brave/components/brave_rewards/core/internal:headers" ]
```

Comment:
similar to the earlier comment making something in `internal` a `public_dep` here seems odd. Also this particular target didn't have a directory structure issue once the process directories were removed so why move it to `internal`?

Reply:
Background on `internal` (currently `core`): 

This is the cross-platform "Rewards engine" code that currently runs in a utility process on desktop/android, but could potentially run in the browser process in the future. It was originally in `third_party/bat-native-ledger` I think.

It seems to me that ideally this folder would be truly internal, and it would only be accessed by iOS or content drivers also in `components/brave_rewards`. If we want to move in that direction, then `internal` as the folder name makes sense, and I can gradually move us there.

Alternatively, we could name this something like "engine".